explainability-service/src/test/java/org/kie/trustyai/service/endpoints/metrics/DisparateImpactRatioEndpointTest.java

```diff
@@ -490,4 +494,31 @@ void listThresholds() {
         assertEquals(0, scheduleList.requests.size());
     }
 
+    @Test
+    void postCorrectFilteringSynthetic() throws JsonProcessingException {
+        datasource.get().reset();
+        Dataframe dataframe = datasource.get().getDataframe(MODEL_ID);
+        List<Prediction> predictions = new ArrayList<>();
+        predictions.add(dataframe.asPredictions().get(0));
+        List<PredictionMetadata> predictionsMetadata = new ArrayList<>();
+        predictionsMetadata.add(new PredictionMetadata("123", "dummy", LocalDateTime.now(), true));
+        dataframe.addPredictions(predictions, predictionsMetadata);
+        datasource.get().saveMetadata(datasource.get().createMetadata(dataframe), MODEL_ID);
+        datasource.get().saveDataframe(dataframe, MODEL_ID);
+
+        final BaseMetricRequest payload = RequestPayloadGenerator.correct();
+
+        final DisparateImpactRatioResponse response = given()
+                .contentType(ContentType.JSON)
+                .body(payload)
+                .when().post()
+                .then()
+                .statusCode(Response.Status.OK.getStatusCode())
+                .extract()
+                .body().as(DisparateImpactRatioResponse.class);
+
+        assertEquals("metric", response.getType());
+        assertEquals("DIR", response.getName());
+        assertFalse(Double.isNaN(response.getValue()));
```

Comment:
@tteofili I don't fully understand the testing for a NaN in DIR (same for the SPD) in this test.
IMO, should the test perhaps be:
- Add valid data
- Calculate DIR
- Add synthetic data
- DIR should be the same?